### PR TITLE
#2Task. dmitry.mukhliada 

### PR DESCRIPTION
**Pull request by N0menIllisLegio**
https://travis-ci.org/github/N0menIllisLegio/db-assignments/builds/712913708

### CODE REVIEW

File: task/mongo-tasks.js [hunk 1]
```diff
@@ -197,7 +374,43 @@ async function task_1_14(db) {
...
+        {
+            $group: {
+                "_id": "$month", 
+                "orders": { "$sum": 1 }
+            }
+        }
+    ]).toArray();
+
+    const result = {
+        "January": ordersInMonths.filter(e => e._id === 1)[0].orders,
```
Comment (IlyaMokin) [inline task/mongo-tasks.js:399]: Looks goon in general, but for the task please try to make all of the operations on DB side.

Comment (N0menIllisLegio) [reply task/mongo-tasks.js]: Done

**Comment by N0menIllisLegio** [comment]
Screen Shot 2020-07-30 at 01 08 44
This is after adding indexes, but when I am trying to add projection stages to pipelines the delay only grows.

This is task_1_22:
`const result = await db.collection('orders').aggregate([
        {
            $project: {
                _id: 0,
                CustomerID: 1,
                OrderID: 1
            }
        },
        {
            $lookup: {
                from: "order-details",
                as: "details",
                let: {orderid: "$OrderID"},
                pipeline: [
                    {
                        $match: {
                            $expr: {
                                $eq: ["$OrderID", "$$orderid"]
                            }
                        }
                    },
                    {
                        $project: {
                            _id: 0,
                            ProductID: 1,
                            UnitPrice: 1
                        }
                    }
                ]

                // localField: "OrderID",
                // foreignField: "OrderID"
            }
        },
        { $unwind: "$details" },
        { $sort: { "details.UnitPrice": -1 } },
        {
            $group: {
                _id: "$CustomerID", 
                product: { $first: "$details.ProductID" },
                price: { $max: "$details.UnitPrice" }
            }
        },
        {
            $lookup: {
                from: "customers",
                as: "customers",

                let: {customerid: "$_id"},
                pipeline: [
                    {
                        $match: {
                            $expr: {
                                $eq: ["$CustomerID", "$$customerid"]
                            }
                        }
                    },
                    {
                        $project: {
                            _id: 0,
                            CompanyName: 1
                        }
                    }
                ]

                // localField: "_id",
                // foreignField: "CustomerID"
            }
        },
        { $unwind: "$customers" },
        {
            $lookup: {
                from: "products",
                as: "products",

                let: {productid: "$product"},
                pipeline: [
                    {
                        $match: {
                            $expr: {
                                $eq: ["$ProductID", "$$productid"]
                            }
                        }
                    },
                    {
                        $project: {
                            _id: 0,
                            ProductName: 1
                        }
                    }
                ]

                // localField: "product",
                // foreignField: "ProductID"
            }
        },
        { $unwind: "$products" },
        {
            $project: {
                _id: 0,
                CustomerID: "$_id",
                ProductName: "$products.ProductName",
                CompanyName: "$customers.CompanyName",
                "PricePerItem": "$price"
            }
        },
        { $sort: { "PricePerItem": -1, CompanyName: 1, ProductName: 1 } }
    ]).toArray();`

Am I doing something wrong?

**Review by IlyaMokin — COMMENTED** [review]
The PR is accepted, closing the PR

**Pull request closed without merging**